core/unit_test/CMakeLists.txt

```diff
@@ -284,6 +284,26 @@ TRIBITS_ADD_EXECUTABLE_AND_TEST(
     TESTONLYLIBS kokkos_gtest ${TEST_LINK_TARGETS}
 )
 
+TRIBITS_ADD_EXECUTABLE_AND_TEST(
+  UnitTest_PushFinalizeHook
+  SOURCES
+    UnitTest_PushFinalizeHook.cpp
+  COMM serial mpi
+  NUM_MPI_PROCS 1
+  FAIL_REGULAR_EXPRESSION "FAILED"
+    TESTONLYLIBS kokkos_gtest ${TEST_LINK_TARGETS}
+)
+
+TRIBITS_ADD_EXECUTABLE_AND_TEST(
+  UnitTest_PushFinalizeHook_terminate
+  SOURCES
+    UnitTest_PushFinalizeHook_terminate.cpp
```

Comment:
@mhoemmen I don't see this file with the tests in the PR, did it get left out?

Reply:
@ndellingwood @ibaned @crtrott I just pushed a commit with the missing files.  Thanks!  Not sure why they didn't transfer over before.  (I started with a Trilinos patch, but the patch didn't merge, so I had to do it by hand.)